gpuflow/dataplane/gpu/cuda/cuda_lpm_factory.cu

```diff
@@ -40,6 +40,15 @@ __global__ void InitIPv6LPMTable(IPv6RuleEntry *ipv6_tbl_24) {
   ipv6_tbl_24[idx].tbl8_ptr = nullptr;
 }
 
+__global__ void InitIPv6TBL8Table(IPv6RuleEntry *ipv6_tbl_8) {
+  int idx = threadIdx.x;
+  ipv6_tbl_8[idx].next_hop = 254;
+  ipv6_tbl_8[idx].valid_flag = false;
+  ipv6_tbl_8[idx].depth = 0;
+  ipv6_tbl_8[idx].external_flag = false;
+  ipv6_tbl_8[idx].tbl8_ptr = nullptr;
+}
+
```

Comment:
Also, to improve the readability, it's time to use template to generate these repetition intialization and setups kernel functions. The type will be inferred when calling kernel functions.

```c++
template <typename IPvxRuleEntry>
__global__ void InitIPvxRuleEntry(IPvxRuleEntry *ipvx_tbl_ptr) {
  int idx = blockIdx.x * blockDim.x + threadIdx.x;
  ipvx_tbl_ptr[idx].next_hop = 254;
  ipvx_tbl_ptr[idx].valid_flag = false;
  ipvx_tbl_ptr[idx].depth = 0;
  ipvx_tbl_ptr[idx].external_flag = false;
  ipvx_tbl_ptr[idx].tbl8_ptr = nullptr;
}
```

Reply:
To Improve consistency, you need to add the tbl8_ptr field on IPv4RuleEntry, but need not to implement it in detailed.